cfbs/commands.py

```diff
@@ -228,24 +229,48 @@ def init_command(index=None, non_interactive=False) -> int:
     """
     CFBSConfig.reload()
 
-    if prompt_user(
-        non_interactive,
-        "Do you wish to build on top of the default policy set, masterfiles? (Recommended)",
-        choices=YES_NO_CHOICES,
-        default="yes",
-    ) in ("yes", "y"):
-        to_add = "masterfiles"
-    else:
-        to_add = prompt_user(
+    branch = None
+    to_add = ""
+    if masterfiles is None:
+        if prompt_user(
             non_interactive,
-            "Specify policy set to use instead (empty to skip)",
-            default="",
-        )
+            "Do you wish to build on top of the default policy set, masterfiles? (Recommended)",
+            choices=YES_NO_CHOICES,
+            default="yes",
+        ) in ("yes", "y"):
+            to_add = "masterfiles"
+        else:
+            to_add = prompt_user(
+                non_interactive,
+                "Specify policy set to use instead (empty to skip)",
+                default="",
+            )
+    elif re.match(r"[0-9]+(\.[0-9]+){2}(\-[0-9]+)?", masterfiles):
+        log.debug("--masterfiles=%s appears to be a version number" % masterfiles)
+        to_add = "masterfiles@%s" % masterfiles
+    elif masterfiles != "no":
+        log.debug("--masterfiles=%s appears to be a branch" % masterfiles)
+        branch = masterfiles
+        to_add = "masterfiles"
```

Comment:
I think there should be a commit here describing the magic happening in this case (and why).

Reply:
I'm guessing you meant comment, not commit?